### PR TITLE
feat: the rspack unsupport inline loader, so we move data-loader to b…

### PR DESCRIPTION
**Pull request by GiveMe-A-Name**
## Description

more detail, see changeset

## Types of changes

- [ ] Docs change / Dependency upgrade
- [ ] Bug fix
- [x] New feature / Improvement
- [ ] Refactoring
- [ ] Breaking change

## Checklist

- [x] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/cli/plugin-data-loader/src/cli/loader.ts [hunk 1]
```diff
@@ -18,7 +18,30 @@ export default async function loader(
...
+      pre[key] = value;
+      return pre;
+    }, {} as Record<string, any>) as { mapFile: string; loaderId?: string };
+
+  // if we can not parse mapFile from resourceQuery, it means the with no need for data-loader handle.
+  if (!options.mapFile) {
+    return source;
+  }
+
+  // const options = this.resourceQuery();
```
Comment (GiveMe-A-Name) [inline packages/cli/plugin-data-loader/src/cli/loader.ts:44]: remove

File: packages/solutions/app-tools/src/builder/shared/builderPlugins/adapterSSR.ts [hunk 2]
```diff
@@ -222,3 +223,12 @@ async function applySSRLoaderEntry<B extends Bundler>(
     }),
   );
 }
+
+function applySSRDataLoader(chain: BundlerChain) {
+  chain.module
+    .rule('ssr-data-loader')
+    .test(/src\/.*\.loader.[t|j]s$/)
```
Comment (yimingjfe) [inline packages/solutions/app-tools/src/builder/shared/builderPlugins/adapterSSR.ts:230]: 有这个配置，可能待看下 https://modernjs.dev/configure/app/source/entries-dir.html

Comment (GiveMe-A-Name) [reply packages/solutions/app-tools/src/builder/shared/builderPlugins/adapterSSR.ts]: 感觉可以不用加这个配置来判断。
src/xxx/xx.loader.ts 已经可以过滤到大部分不匹配的文件。
如果剩余的，在 data-loader 有进行判断。如果拿不到 mapFile 的 query 会直接 return source .

Comment (GiveMe-A-Name) [reply packages/solutions/app-tools/src/builder/shared/builderPlugins/adapterSSR.ts]: Done~

**Review by yimingjfe — APPROVED** [review]

**Pull request merged**